x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -175,6 +183,12 @@ export class Embeddable
     }
   }
 
+  private updateActiveData = (data: unknown, inspectorAdapters?: Adapters | undefined) => {
+    if (inspectorAdapters?.tables) {
+      this.activeData = inspectorAdapters.tables;
```

Comment:
Please use `this.activeData = inspectorAdapters;` here and implement `hasDatatableContent` using `inspectorAdapters.tables && Object.keys(inspectorAdapters).length > 0` - that what I meant with my note in a previous comment:

> Note: This means the Lens embeddable should just return the complete adapters object instead of unwrapping .tables in https://github.com/elastic/kibana/pull/83654/files#diff-0bf3b6d3425415123682762f7cfa707621c3a900b643231fd327f365286bd29dR188

The key `tables` in the adapters object is part of the API. There might be other inspector adapters recording completely different things (e.g. request/response payloads), this is not just about tables.